### DISCUSSION TITLE
GraphHopper offline data folder

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#21] [Q&A] [answered]
Menu `Routing | Download graphs` allows you to download GraphHopper data for offline use. I noticed that the .ZIP file is being downloaded to the default `/Android/data/gr.talent.cruiser/files/graphs` but wanting to open Graph in the application, it is looking for in the folder `/Android/media/gr.talent.cruiser` (so subfolder **Data** vs **Media**). I don't think it's possible to change the location, but is the GraphHopper data needed at all since BRouter gives more? As far as I remember, BRouter doesn't give street names what GraphHopper does (regardless of likely routing nuances).

**Comment by devemux86** [comment, marked as answer]
Cruiser downloads the graph zip files and unzip them to the location selected in "Settings | Application | Storage folder".
You can activate the GraphHopper graphs in the **Local** tab of the download manager (no need to open them manually).
(the downloads are automatically activated / loaded)

> is the GraphHopper data needed at all since BRouter gives more?

- GraphHopper (offline) provides country sized routing data. Graphs (like maps) use the [Geofabrik](https://download.geofabrik.de/) regions.
GraphHopper routing data contain speed limits and street names.
- BRouter (offline) provides cross border routing and powerful customized [profiles](https://brouter.de/brouter/profiles2/).
BRouter routing data do not contain speed limits and street names.
https://github.com/abrensch/brouter/issues/55

**Reply by Ciutnik** [reply]
Thanks for the detailed instructions! I didn't realize that the downloaded Graph needs to be activated! (It is possible that I deactivated it myself during intensive play with the application...) 

Sorry for the confusion and thanks again for your reply.

Edit: i see that the request for street names in BRouter was made in 2016. I see it badly...

**Reply by devemux86** [reply]
> I didn't realize that the downloaded Graph needs to be activated!

The downloaded files are automatically activated / loaded (and you can change them later in "Local" tab).

In the case of graphs, only one graph can be active / loaded at a time.
In the case of maps, multiple maps can be active / loaded simultaneously.

**Reply by Ciutnik** [reply]
Of course everything works as it should! 
I probably suggested the `Open graph` menu (which opens a selection of files in the `Media/...` subfolder, regardless of the application's system settings) :-)
Thanks!

**Reply by devemux86** [reply]
> I probably suggested the Open graph menu

Open graph (and open poi) will be removed in next version to avoid confusion and as they are not needed anymore.

We can  keep the open map / theme (in advanced menu) to load maps and themes available from external sources.

**Reply by Ciutnik** [reply]
> We can keep the open map / theme

This would be good (for me personally) as I would like to use a theme I modified from a previous application.  I added the visibility of museums, ruins and churches because I like to see them as landmarks. This is my old habit from the days of paper maps.
